ovs/lib/vdisk.py

```diff
@@ -527,65 +541,73 @@ def create_new(diskname, size, storagedriver_guid):
     @celery.task(name='ovs.vdisk.create_volume')
     def create_volume(location, size, storagerouter_guid=None):
         """
-        Create a volume using filesystem calls
-        Calls "truncate" to create sparse raw file
+        Create a volume
+        !! This method is for compatibility with the cinder driver
+        !! Other callers should use VDiskController.create_new
 
-        @param location: location, filename
-        @param size: size of volume, GB
-        @param: storagerouter_guid: use ssh client to create file on remote storagerouter
-        @return None
+        :param location: location, filename
+        :param size: size of volume, GB
+        :param: storagerouter_guid: create file on remote storagerouter
         """
         logger.info('Creating volume {0} of {1} GB'.format(location, size))
 
-        client = SSHClient(StorageRouter(storagerouter_guid)) if storagerouter_guid is not None else SSHClient('127.0.0.1')
-        if client.file_exists(location):
-            raise RuntimeError('File already exists at %s' % location)
-
-        output = client.run('truncate -s {0}G "{1}"'.format(size, location)).strip()
+        if storagerouter_guid is not None:
+            storagerouter = StorageRouter(storagerouter_guid)
+        else:
+            storagerouter = System.get_my_storagerouter()
+        for storagedriver in storagerouter.storagedrivers:
+            if location.startswith(storagedriver.mountpoint):
```

Comment:
startswith is not a very smart way of checking
Eg:
* Mountpoint /mnt/mtpt_1
* Mountpoint /mnt/mtpt_10

if location is /mnt/mtpt_1 and iterator first enters the storagedriver with mountpoint /mnt/mtpt_10, you'll end up creating the volume on the wrong storagedriver